File: Code/src/presentation/theme.py
import flet as ft

BG_PAGE      = "#E8EDF0"
BG_CARD      = "#FFFFFF"
BORDER       = "#DDE6EA"
ACCENT       = "#1A7F89"
ACCENT_DARK  = "#0F6B73"
ACCENT_LIGHT = "#2A9D97"
TEXT_PRIMARY = "#0A3136"
TEXT_MUTED   = "#4E6A70"
URGENT_CRIT  = "#C0392B"
URGENT_HIGH  = "#E67E22"
URGENT_MED   = "#F39C12"

RADIUS_CARD  = 28
RADIUS_BTN   = 24
RADIUS_CHIP  = 14
RADIUS_IMG   = 22

PHONE_WIDTH  = 390
PHONE_HEIGHT = 720
CARD_WIDTH   = 370
BTN_WIDTH    = 330
BTN_HEIGHT   = 44
IMG_HEIGHT   = 256

def card_shadow() -> list:
    return []

def elevated_shadow() -> list:
    return []

def btn_style(radius: int = RADIUS_BTN) -> ft.ButtonStyle:
    return ft.ButtonStyle(shape=ft.RoundedRectangleBorder(radius=radius), padding=12)

def urgency_color(urgencia: str) -> str:
    return {"critica": URGENT_CRIT, "alta": URGENT_HIGH}.get(urgencia.lower(), URGENT_MED)

File: Code/lib/presentation/Theme.dart
import 'package:flutter/material.dart';

class AppTheme {
  static const bgPage = Color(0xFFE8EDF0);
  static const bgCard = Color(0xFFFFFFFF);
  static const border = Color(0xFFDDE6EA);
  static const accent = Color(0xFF1A7F89);
  static const accentDark = Color(0xFF0F6B73);
  static const accentLight = Color(0xFF2A9D97);
  static const textPrimary = Color(0xFF0A3136);
  static const textMuted = Color(0xFF4E6A70);
  static const urgentCrit = Color(0xFFC0392B);
  static const urgentHigh = Color(0xFFE67E22);
  static const urgentMed = Color(0xFFF39C12);

  static const double radiusCard = 28.0;
  static const double radiusBtn = 24.0;
  static const double radiusChip = 14.0;
  static const double radiusImg = 22.0;
  static const double imgHeight = 256.0;
  static const double btnHeight = 44.0;
  static const double phoneWidth = 390.0;

  static BoxDecoration cardDecoration() => BoxDecoration(
        color: bgCard,
        borderRadius: BorderRadius.circular(radiusCard),
        boxShadow: [
          BoxShadow(
            color: Colors.black.withValues(alpha: 0.08),
            blurRadius: 8,
            offset: const Offset(0, 2),
          ),
        ],
      );

  static ButtonStyle elevatedButtonStyle(Color bgColor) =>
      ElevatedButton.styleFrom(
        backgroundColor: bgColor,
        foregroundColor: Colors.white,
        disabledBackgroundColor: bgColor.withValues(alpha: 0.45),
        disabledForegroundColor: Colors.white.withValues(alpha: 0.9),
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(radiusBtn)),
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 0),
        minimumSize: const Size(double.infinity, btnHeight),
      );

  static ButtonStyle headerButtonStyle(Color bgColor) =>
      ElevatedButton.styleFrom(
        backgroundColor: bgColor,
        foregroundColor: Colors.white,
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(radiusBtn)),
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 0),
        minimumSize: const Size(0, btnHeight),
      );

  static ThemeData themeData() => ThemeData(
        useMaterial3: true,
        scaffoldBackgroundColor: bgPage,
        appBarTheme: const AppBarTheme(
          backgroundColor: bgCard,
          surfaceTintColor: Colors.transparent,
        ),
        scrollbarTheme: ScrollbarThemeData(
          thumbVisibility: WidgetStateProperty.all(true),
          thickness: WidgetStateProperty.all(8),
          radius: const Radius.circular(8),
          thumbColor: WidgetStateProperty.all(accent.withValues(alpha: 0.38)),
          trackColor: WidgetStateProperty.all(accent.withValues(alpha: 0.08)),
        ),
      );
}
